shared/src/commonMain/kotlin/org/hyperskill/app/core/domain/repository_cache/RepositoryCacheProxy.kt

```diff
@@ -36,7 +36,7 @@ class RepositoryCacheProxy<in Key : Any, Value : Any?>(
                 } else {
                     loadValuesAndUpdateCache(keysToLoadFromRemote.toList())
                         .map { remoteValues ->
-                            cachedValues + remoteValues
+                            (cachedValues + remoteValues).sortedBy(getKeyFromValue)
```

Comment:
@XanderZhu I mean sorting by parameter `keys: List<Key>`, `getValues(keys = [2, 5, 1, 7])` -> result list sorted in order of `[2, 5, 1, 7]`

For example [stepik-ios implementation](https://github.com/StepicOrg/stepik-ios/blob/master/Stepic/Sources/Extensions/Foundation/Array%2BReordering.swift)

Reply:
I'll ask the backend team about do they sorting for every endpoint when getting objects by ids or not. If not then we don't need sorting here.